src/classes/VOL_CTRL_SendBulkEmail.cls

```diff
@@ -118,28 +119,113 @@ public with sharing class VOL_CTRL_SendBulkEmail {
     }
     
     public string templateObject { get; set; }
-    
-    // action method to load up the email template the user has chosen
-    public PageReference LoadEmailTemplate() {
+
+    /*******************************************************************************************************************
+    * @description Sets the email body and subject from the template matching the template Id selected
+    * @return null
+    */
+    public PageReference loadEmailTemplate() {
+        isHtmlTemplate = false;
+        strSubject = null;
+        strMessage = null;
+
+        if (templateId == null) {
+            return null;
+        }
+
         try {
-            if (templateId != null) {
-                // Ensure the user has access to the object and fields before querying
-                UTIL_Describe.checkReadAccess('EmailTemplate', new Set<String>{'Id','Name', 'Subject', 'HtmlValue', 'Body'});    
-                    
-                EmailTemplate et = [select Id, Name, Subject, HtmlValue, Body from EmailTemplate where Id = :templateId];
-                strSubject = et.Subject;
-                strMessage = (et.HtmlValue != null) ? et.HtmlValue : et.Body;
+            // Ensure the user has access to the object and fields before querying
+            UTIL_Describe.checkReadAccess(
+                'EmailTemplate',
+                new Set<String>{ 'Id', 'Name', 'Subject', 'HtmlValue', 'Body' }
+            );
+
+            EmailTemplate template = [
+                SELECT Id, Name, Subject, HtmlValue, Body, TemplateType
+                FROM EmailTemplate
+                WHERE Id = :templateId
+            ];
+
+            strSubject = template.Subject;
+
+            if (template.HtmlValue != null) {
+                isHtmlTemplate = true;
+                strMessage = cleanHtml(template);
+
             } else {
-                strSubject = null;
-                strMessage = null;      
+                strMessage = template.Body;
             }
+
             return null;
+
         } catch (Exception e) {
-            ApexPages.addMessages(e);  
+            ApexPages.addMessages(e);
             return null;
         }
     }
 
+    /*******************************************************************************************************************
+    * @description Html values are only cleaned automatically by Salesforce when the TemplateType is html
+    * in order to clean the other html values, we will do a temporary commit to a Rich Text Area field to allow
+    * Salesforce to strip the non whitelisted tags for us
+    * @return Volunteer_Job__c the job associated with this page
+    */
+    @TestVisible
+    private String cleanHtml(EmailTemplate template) {
```

Comment:
What do you think of a method name such as `getCleanHtmlFromTemplate`?

Reply:
@force2b Not sure about the name suggestion. I don't think we are just getting something we are cleaning it ourselves. Similar to escapeHtml methods on the String class. Should I use escapeHtml here instead?

Reply:
I guess my thought there is that if we were going to overload the method and take something else, the parameter itself explains that part. What do you think about escapeHtmlFrom(EmailTemplate template) ?

Reply:
Maybe we should look at the entire process differently. In the `loadEmailTemplate` method, how about replacing the block that calls `cleanHtml` with the following
```
isHtmlTemplate = (template.HtmlValue != null);
messageBody = cleanTemplateMessage(template);
```

Then the `cleanHtml` method gets renamed to `cleanTemplateMessage` and it handles either returning the correct content (`template.Body` or cleaned html). (Note that I'm changing `strMessage` to `messageBody` just to avoid the Hungarian notation).

Reply:
@jjbennett Did you see the comment above?

Reply:
No sorry missed that one, looking now.

Reply:
I can try and rework it a bit yet, I'll ping you when I have settled on something. Thanks!